server/memphis_handlers_integrations.go

```diff
@@ -171,153 +137,53 @@ func (it IntegrationsHandler) UpdateIntegration(c *gin.Context) {
-		filter := bson.M{"name": "slack"}
-		err := integrationsCollection.FindOne(context.TODO(), filter).Decode(&integrationFromDb)
+		integrationType := strings.ToLower(body.Name)
+		keys, properties := createIntegrationsKeysAndProperties(integrationType, "", "", false, false, false, body.Keys["access_key"], body.Keys["secret_key"], body.Keys["bucket_name"], body.Keys["region"])
+		s3Integration, err := updateS3Integration(keys, properties)
 		if err != nil {
-			return slackIntegration, err
+			serv.Errorf("updateS3Integration: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
 		}
-		authToken = integrationFromDb.Keys["auth_token"]
-	}
-
-	err := testSlackIntegration(authToken, channelID, "Slack integration with Memphis was updated successfully")
-	if err != nil {
-		return slackIntegration, err
-	}
-	keys, properties := createSlackKeysAndProperties(authToken, channelID, pmAlert, svfAlert, disconnectAlert, uiUrl)
-	filter := bson.M{"name": "slack"}
-	err = integrationsCollection.FindOneAndUpdate(context.TODO(),
-		filter,
-		bson.M{"$set": bson.M{"keys": keys, "properties": properties}}).Decode(&slackIntegration)
-	if err == mongo.ErrNoDocuments {
-		slackIntegration = models.Integration{
-			ID:         primitive.NewObjectID(),
-			Name:       "slack",
-			Keys:       keys,
-			Properties: properties,
+		integration = s3Integration
+		if integration.Keys["secret_key"] != "" {
+			lastCharsSecretKey := integration.Keys["secret_key"][len(integration.Keys["secret_key"])-4:]
+			integration.Keys["secret_key"] = "****" + lastCharsSecretKey
 		}
-		integrationsCollection.InsertOne(context.TODO(), slackIntegration)
-	} else if err != nil {
-		return slackIntegration, err
-	}
-
-	integrationToUpdate := models.CreateIntegrationSchema{
-		Name:       "slack",
-		Keys:       keys,
-		Properties: properties,
-		UIUrl:      uiUrl,
-	}
-
-	msg, err := json.Marshal(integrationToUpdate)
-	if err != nil {
-		return slackIntegration, err
-	}
-	err = serv.sendInternalAccountMsgWithReply(serv.GlobalAccount(), INTEGRATIONS_UPDATES_SUBJ, _EMPTY_, nil, msg, true)
-	if err != nil {
-		return slackIntegration, err
-	}
-
-	update := models.ConfigurationsUpdate{
-		Type:   sendNotificationType,
-		Update: svfAlert,
-	}
-	serv.SendUpdateToClients(update)
-
-	slackIntegration.Keys = keys
-	slackIntegration.Properties = properties
-	return slackIntegration, nil
-}
 
-func testSlackIntegration(authToken string, channelID string, message string) error {
-	slackClientTemp := slack.New(authToken)
-	_, err := slackClientTemp.AuthTest()
-	if err != nil {
-		return errors.New("Invalid auth token")
-	}
-	attachment := slack.Attachment{
-		AuthorName: "Memphis",
-		Text:       message,
-		Color:      "#6557FF",
+	default:
+		serv.Warnf("CreateIntegration: Unsupported integration type - " + body.Name)
+		c.AbortWithStatusJSON(400, gin.H{"message": "CreateIntegration: Unsupported integration type - " + body.Name})
```

Comment:
CreateIntegration: ... this kind of message is for us to log not for the users.
users don't care about the function names.
so omit it please